### PR TITLE
Feature retry processor (v2)

### PR DESCRIPTION
**Pull request by daviddahl**
(This is a refactored version after the EffectHandler, etc changes in #532)

Add RetryProcessor with Testing and Benchmarks

Summary

Adds a new RetryProcessor component that provides reliable message delivery with exponential backoff retry logic for the OTAP dataflow engine.

Changes

•  New RetryProcessor: Handles message retries with configurable policies, ACK/NACK feedback, and graceful shutdown
•  9 comprehensive unit tests: Cover all major functionality including retry logic, backpressure, and edge cases  
•  Performance benchmarks: Microbenchmarks showing ~150-230ns latency for core operations

Key Features

•  Exponential backoff with configurable limits (default: 3 retries, 1s initial delay)
•  ACK/NACK message tracking with HashMap-based O(1) operations
•  Queue limits for backpressure handling
•  Dynamic configuration updates
•  Graceful shutdown with pending message flushing

### CODE REVIEW

File: rust/otap-dataflow/crates/engine/src/retry_processor.rs [hunk 1]
```diff
@@ -0,0 +1,692 @@
...
+                log::debug!("Scheduled message {id} for retry attempt {retry_count}");
+            } else {
+                log::error!(
+                    "Message {} exceeded max retries ({}), dropping. Last error: {}",
+                    id,
+                    self.config.max_retries,
+                    pending.last_error
+                );
+            }
+        }
```
Comment (albertlockett) [inline rust/otap-dataflow/crates/engine/src/retry_processor.rs:178]: nit: I noticed that in the `acknowledge` method, we warn if the map doesn't contain the `id`. Should we do the same thing here?
```suggestion
        } else {
            log::warn!("Attempted to handle nack for non-existent message with ID: {id}");
        }
```

**Review by albertlockett — APPROVED** [review]
LGTM. Nice work @daviddahl !
